s3/src/test/scala/docs/scaladsl/S3SourceSpec.scala

```diff
@@ -167,29 +161,28 @@ class S3SourceSpec extends S3WireMockBase with S3ClientIntegrationSpec {
     val versionId = "3/L4kqtJlcpXroDTDmJ+rmSpXd3dIbrHY+MTRCxf3vjVBH40Nr8X8gdRQBpUMLUo"
     mockDownloadSSECWithVersion(versionId)
 
-    val downloadResult =
-      S3.download(bucket, bucketKey, versionId = Some(versionId), sse = Some(sseCustomerKeys))
+    val s3Source =
+      S3.getObject(bucket, bucketKey, versionId = Some(versionId), sse = Some(sseCustomerKeys))
 
-    val Some((s3Source: Source[ByteString, _], metadata)) =
-      downloadResult.runWith(Sink.head).futureValue: @nowarn("msg=match may not be exhaustive")
-    val result = s3Source.map(_.utf8String).runWith(Sink.head)
+    val (metadata, result) = s3Source.map(_.utf8String).toMat(Sink.head)(Keep.both).run()
 
     result.futureValue shouldBe bodySSE
-    metadata.versionId.fold(fail("unable to get versionId from S3")) { vId =>
+    metadata.futureValue.versionId.fold(fail("unable to get versionId from S3")) { vId =>
       vId shouldEqual versionId
     }
   }
 
-  it should "fail if request returns 404" in {
+  it should "throw the correct S3Exception if a request returns 404" in {
 
     mock404s()
 
     val download = S3
-      .download("nonexisting-bucket", "nonexisting_file.xml")
+      .getObject("nonexisting-bucket", "nonexisting_file.xml")
       .runWith(Sink.head)
-      .futureValue
 
-    download shouldBe None
+    download.failed.futureValue should matchPattern {
```

Comment:
So even though it can be argued that this test is not necessary anymore since we aren't wrapping the 404 case into an `Option`, I rewrote it to check for the same exception just to make sure its behaving the same way.